src/warnet/tank.py

```diff
@@ -112,6 +114,12 @@ def exporter_name(self):
     def status(self) -> RunningStatus:
         return self.warnet.container_interface.get_status(self.index, ServiceType.BITCOIN)
 
+    @property
+    def autonomous_system(self):
+        if self._a_system is None:
+            self._a_system = generate_as(self.warnet)
+        return self._a_system
+
```

Comment:
Personally, I'm not a huge fan of passing the warnet object in to tanks, especially in this case as it's having it's `a_systems` property updated by the tank.

IMHO if the warnet object needs to keep track of what's been assigned then the logic belongs with the warnet object to assign the tanks a autonomous_system number.